rustfft.wai

```diff
@@ -0,0 +1,70 @@
+enum fft-direction {
+    forward,
+    inverse
+}
+
+resource complex {
+    static new-f64: func(re: float64, im: float64) -> complex
+    i: func() -> complex
+}
+
+resource inner-fft {
+    process-with-scratch: func(buffer: list<complex>, scratch: list<complex>)
+    process-outofplace-with-scratch: func(input: list<complex>, output: list<complex>, scratch: list<complex>)
+    get-inplace-scratch-len: func() -> u32
+    get-outofplace-scratch-len: func() -> u32
+    process: func(buffer: list<complex>)
+}
+
+resource fft-planner {
+    static new-fft-planner: func() -> fft-planner
```

Comment:
Let's just call this `new()`, otherwise you would call it as `FftPlanner.NewFftPlanner()`, which seems like unnecessary repetition.

Reply:
refactored the `fft-planner` with a `new` initializer